tests/Fixtures/Populator/PersonNamePopulator.php

```diff
@@ -16,8 +16,11 @@ class PersonNamePopulator implements Populator
 {
     public function populate(object $target, object $source, ?object $ctx = null): void
     {
-        $separator = $ctx?->getValue('separator') ?? ' ';
+        $separator = ' ';
+        if ($ctx?->hasKey('separator')) {
+            $separator = $ctx->getValue('separator');
+        }
 
-        $target->setFullName(implode($separator, [$source->getFirstname(), $source->getLastname()]));
+        $target->setFullName(sprintf("%s%s%s", $source->getFirstname(), $separator, $source->getLastname()));
```

Comment:
Not very future-proof, if you want to e.g. add the middle name or a title.

With the current code, you just need to add them to the array: 
```php
implode($separator, [
    $source->getTitle(), 
    $source->getFirstname(), 
    $source->getMiddlename(), 
    $source->getLastname(),
])
```
With your proposal, it would be either:
```php
sprintf(
    "%s%s%s%s%s%s%s", 
    $source->getTitle(), 
    $separator, 
    $source->getFirstname(),
    $separator,
    $source->getMiddlename(), 
    $separator,
    $source->getLastname(),
)
```
or maybe: 
```php
sprintf(
    "%2$s%1$s%3$s%1$s%4$s%1$s%5$s", 
    $separator, 
    $source->getTitle(), 
    $source->getFirstname(), 
    $source->getMiddlename(), 
    $source->getLastname()
)
```
Both are pretty unreadable if you ask me.

Reply:
I can agree with that.
And yes, both are unreadable anyway better that pure concatention.